### PR TITLE
Added "source" taxonomy for external stories

### PR DESCRIPTION
**Pull request by rjucf**

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -1063,7 +1063,12 @@ function display_news_schema( $post ) {
 function display_external_stories_list_item( $story_id, $show_description ) {
 	$story_url = get_post_meta( $story_id, 'externalstory_url', True );
 	$story_text = get_post_meta( $story_id, 'externalstory_text', True );
-	$story_source = get_post_meta( $story_id, 'externalstory_source', True );
+	$source_name = wp_get_post_terms( $story_id, 'sources' )[0]->name;
```
Comment (jmbarne3) [inline functions.php]: If this doesn't return anything, it's potentially going to blow up for trying to access a property on a null object, and/or an out of range index exception.

File: custom-taxonomies.php [hunk 2]
```diff
@@ -133,5 +133,30 @@ class Groups extends CustomTaxonomy{
...
+		$all_times			= 'All Sources',
+		$parent_item		= 'Parent Source',
+		$parent_item_colon	= 'Parent : Source',
+		$edit_item			= 'Edit Source',
+		$update_item		= 'Update Source',
+		$add_new_item		= 'Add New Source',
+		$new_item_name		= 'New Source Name',
+		$menu_name			= NULL,
+		$hierarchical		= True;
+}
```
Comment (jmbarne3) [inline custom-taxonomies.php:159]: Some weird tabbing in the above block.

**Review by jmbarne3 — APPROVED** [review]

**Review by cadie — APPROVED** [review]

File: custom-taxonomies.php [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
+
 		// Do not register the taxonomy with the post type here.
 		// Register it on the `taxonomies` attribute of the post type in
 		// custom-post-types.php
-		$object_type	= Array(), 
-		
-		$general_name		    = 'Post Tags',
+		$object_type	= Array(),
+
+		$general_name		= 'Post Tags',
```
Comment (cjg89) [inline custom-taxonomies.php]: Mixed tabs + spaces here.

File: functions/api.php [hunk 4]
```diff
@@ -33,11 +33,11 @@ public static function register_rest_routes() {
          */
         public static function get_external_stories( $request ) {
             // Handle args and set defaults
-            $search     = $request['search'];
+			$search     = $request['search'];
```
Comment (cjg89) [inline functions/api.php]: Mixed tabs/spaces

File: functions/api.php [hunk 5]
```diff
@@ -102,15 +102,19 @@ private static function prepare_external_story_for_response( $post, $request ) {
...
                 'publish_date' => '',
                 'categories'   => array()
             );
 
             $retval['title']        = $post->post_title;
             $retval['link_text']    = get_post_meta( $post->ID, 'externalstory_text', true );
             $retval['description']  = get_post_meta( $post->ID, 'externalstory_description', true );
-            $retval['url']          = get_post_meta( $post->ID, 'externalstory_url', true );
-            $retval['source']       = get_post_meta( $post->ID, 'externalstory_source', true );
+			$retval['url']          = get_post_meta( $post->ID, 'externalstory_url', true );
```
Comment (cjg89) [inline functions/api.php]: Mixed tabs/spaces

File: functions/api.php [hunk 6]
```diff
@@ -102,15 +102,19 @@ private static function prepare_external_story_for_response( $post, $request ) {
...
+			$retval['source']       = get_post_meta( $post->ID, 'externalstory_source', true );
+			$retval['source_name']  = wp_get_post_terms( $post->ID, 'sources' )[0]->name;
+            $retval['source_image'] = get_field( 'news_source_image', wp_get_post_terms( $post->ID, 'sources' )[0] )['url'];
```
Comment (cjg89) [inline functions/api.php]: line 117:  `wp_get_post_terms( $post->ID, 'sources' )[0]` will throw at least a notice if `wp_get_post_terms()` doesn't return at least one item, or if it returns `WP_Error`.

Trying to access `get_field(...)['url']` may also throw similar notices if a valid value isn't returned.

File: functions/api.php [hunk 7]
```diff
@@ -102,15 +102,19 @@ private static function prepare_external_story_for_response( $post, $request ) {
...
             );
 
             $retval['title']        = $post->post_title;
             $retval['link_text']    = get_post_meta( $post->ID, 'externalstory_text', true );
             $retval['description']  = get_post_meta( $post->ID, 'externalstory_description', true );
-            $retval['url']          = get_post_meta( $post->ID, 'externalstory_url', true );
-            $retval['source']       = get_post_meta( $post->ID, 'externalstory_source', true );
+			$retval['url']          = get_post_meta( $post->ID, 'externalstory_url', true );
+			$retval['source']       = get_post_meta( $post->ID, 'externalstory_source', true );
+			$retval['source_name']  = wp_get_post_terms( $post->ID, 'sources' )[0]->name;
```
Comment (cjg89) [inline functions/api.php]: line 116: same issue Jim noted in functions.php.

**Review by cjg89 — CHANGES_REQUESTED** [review]

**Review by cjg89 — APPROVED** [review]

**Pull request merged**